bach/tests/functional/bach/test_df_materialize.py

```diff
@@ -1,19 +1,23 @@
 """
 Copyright 2021 Objectiv B.V.
 """
+from typing import Union
 from unittest.mock import ANY
 
 import pytest
+from sqlalchemy.engine import Engine
 
 from bach import SeriesUuid
 from sql_models.graph_operations import get_graph_nodes_info
+from sql_models.model import Materialization
 from sql_models.util import is_bigquery, is_postgres
 from tests.functional.bach.test_data_and_utils import assert_equals_data, get_df_with_test_data, \
     get_df_with_json_data
 
 
 @pytest.mark.parametrize("inplace", [False, True])
-def test_materialize(inplace: bool, engine):
+@pytest.mark.parametrize("materialization", [Materialization.CTE, 'temp_table'])
```

Comment:
Bit strange to mix constant and string notation here unless you really want to test both here, but in that case you should do all variations?
```suggestion
@pytest.mark.parametrize("materialization", [Materialization.CTE,Materialization.TEMP_TABLE])
```

Reply:
Yeah I wanted to test both cte and temp-table, and both string and enum parameters. I figured this is a nice trade-off between test coverage and not completely exploding the number of tests